bionic/module_reloader.py

```diff
@@ -0,0 +1,24 @@
+import builtins
+from importlib import reload
+from sys import modules as module_registry
+
+
+def recursive_reload(module):
+    original_import = builtins.__import__
+    already_reloaded = set()
+
+    def custom_import(name, globals=None, locals=None, fromlist=[], level=0):
+        if name in module_registry:
+            module = module_registry[name]
+            if name not in already_reloaded and hasattr(module, "__file__"):
+                already_reloaded.add(name)
+                try:
+                    reload(module)
+                except Exception:
```

Comment:
What exceptions are we catching here? In general, if we're going to swallow all exceptions, we should probably have a comment explaining why.